crankshaft-engine/src/service/runner/backend/generic.rs

```diff
@@ -168,7 +184,19 @@ impl crate::Backend for Backend {
 
                         loop {
                             let monitor = config.resolve_monitor(&substitutions).unwrap();
-                            let output = driver.run(monitor).await.unwrap();
+
+                            let result = select! {
+                                _ = token.cancelled() => {
+                                    Err(eyre!("task has been cancelled"))
+                                }
+                                res = driver.run(monitor) => {
+                                    res
+                                }
+                            };
+
+                            // TODO: does there need to be a cleanup mechanism that gets invoked due
```

Comment:
This is also a question for the `select!`s in `crankshaft-engine/src/service/runner/backend/docker.rs`. Does this need to be handled before merge?

Reply:
Actually, we already have a "kill" mechanism within the generic driver right now, but it's unused. I think this is where it should be invoked, so I'll add it.